kubeflow/modeldb/modeldb.libsonnet

```diff
@@ -0,0 +1,319 @@
+{
+  all(params, namespace):: [
+    $.parts(params, namespace).backendService,
+    $.parts(params, namespace).backendPVC,
+    $.parts(params, namespace).backendDeployment,
+    $.parts(params, namespace).dbService,
+    $.parts(params, namespace).dbPVC,
+    $.parts(params, namespace).dbDeployment,
+    $.parts(params, namespace).frontendService,
+    $.parts(params, namespace).frontendDeployment,
+  ],
+
+  parts(params, namespace):: {
+    backendService: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        name: "modeldb-backend",
+        namespace: namespace,
+        labels: {
+          app: "modeldb",
+          component: "backend",
+        },
+      },
+      spec: {
+        type: "ClusterIP",
+        ports: [
+          {
+            port: 6543,
+            protocol: "TCP",
+            name: "api",
+          },
+        ],
+        selector: {
+          app: "modeldb",
+          component: "backend",
+        },
+      },
+    },  // backendService
+    backendPVC: {
+      apiVersion: "v1",
+      kind: "PersistentVolumeClaim",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "backend",
+        },
+        name: "modeldb-backend-pvc",
+        namespace: namespace,
+      },
+      spec: {
+        accessModes: [
+            "ReadWriteOnce",
+        ],
+        resources: {
+          requests: {
+            storage: params.modeldbBackendPvcSize
+          },
+        },
+      },
+    }, // backendPersistentVolumeClaim
+    backendDeployment: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "backend",
+        },
+        name: "modeldb-backend",
+        namespace: namespace,
+      },
+      spec: {
+        replicas: 1,
+        template: {
+          metadata: {
+            labels: {
+              app: "modeldb",
+              component: "backend",
+            },
+            name: "modeldb-backend",
+          },
+          spec: {
+            containers: [
+              {
+                env: [
+                  {
+                    name: "MONGODB_HOST",
```

Comment:
could you make this a param? Also make mongodb deployment optional. Some public clouds or on-prems have dedicated mongodb clusters that are replicated and it'd be good to allow to easily use them

Reply:
Fixed to allow MONGODB_HOST to be specified in the params.